numpy/f2py/rules.py

```diff
@@ -211,7 +205,7 @@
 \tPy_TYPE(&PyFortran_Type) = &PyType_Type;
 \timport_array();
 \tif (PyErr_Occurred())
-\t\t{PyErr_SetString(PyExc_ImportError, \"can't initialize module #modulename# (failed to import numpy)\"); return RETVAL;}
```

Comment:
This looks like it should be `return NULL`, but let's leave that to a follow-up

Reply:
this pattern repeats in a number of places I have a draft pull to try something new..